tests/network/test_failure_modes.py

```diff
@@ -117,3 +125,62 @@ def test_node_has_changed_cert(federated_alice, federated_ursulas):
 
     # Cool - we didn't crash because of SSLError.
     # TODO: Assertions and such.
+
+
+def test_huge_treasure_maps_are_rejected(federated_alice, federated_ursulas):
+    federated_alice.network_middleware = EvilMiddleWare()
+
+    firstula = list(federated_ursulas)[0]
+
+    ok_amount = 10 * 1024  # 10k
+    ok_data = os.urandom(ok_amount)
+
+    with pytest.raises(BytestringSplittingError):
```

Comment:
Curious - What specifically causes `BytestringSplittingError` to be raised here?

Reply:
Basically, the occurrence of a BytestringSplittingError means that the decorator that prevents us from passing in overlength data is allowing it through, and we are now in the view logic.

The view logic currently doesn't validate anything so it just throws whatever comes in into the ByteStringSplitter.
In this case the bytes are some random junk, so we get a BytestringSplittingError.

I'm not testing PRE, I'm just testing the decorator on the view here so catching a BytestringSplittingError is a fine test of that.